core/src/main/scala/org/apache/spark/TaskEndReason.scala

```diff
@@ -90,6 +90,10 @@ case class FetchFailed(
  *
  * `fullStackTrace` is a better representation of the stack trace because it contains the whole
  * stack trace including the exception and its causes
+ *
+ * `preserveCause` is used to keep the exception itself so it is available to the
+ * driver. This may be set to `false` in the event that the exception is not in fact
+ * serializable.
```

Comment:
I think this is the wrong place for this doc on `preserveCause`.  That should go on the constructor that has it.  Here I'd just doc the `exception` (and add an explanation of which exception is corresponds to as Patrick asked)